bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallHandler.java

```diff
@@ -0,0 +1,187 @@
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to Tesla Powerwall", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to Tesla Powerwall", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() throws IOException {
+
+        TeslaPowerwallConfiguration config = getConfigAs(TeslaPowerwallConfiguration.class);
+        if (config.email != null && config.password != null) {
+            String token = webTargets.getToken(config.email, config.password);
+            Operations operations = webTargets.getOperations(token);
+            if (operations != null) {
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_MODE,
+                        new StringType(operations.mode));
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_RESERVE,
+                        new QuantityType<>(((operations.reserve / 0.95) - (5 / 0.95)), Units.PERCENT));
+            }
+
+            BatterySOE batterySOE = webTargets.getBatterySOE(token);
+            GridStatus gridStatus = webTargets.getGridStatus(token);
+            SystemStatus systemStatus = webTargets.getSystemStatus(token);
+            MeterAggregates meterAggregates = webTargets.getMeterAggregates(token);
+
+            updateStatus(ThingStatus.ONLINE);
+            if (batterySOE != null) {
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_BATTERYSOE,
+                        new QuantityType<>(((batterySOE.soe / 0.95) - (5 / 0.95)), Units.PERCENT));
+            }
+            if (gridStatus != null) {
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_GRIDSTATUS,
+                        new StringType(gridStatus.grid_status));
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_GRIDSERVICES,
+                        (gridStatus.grid_services ? OnOffType.ON : OnOffType.OFF));
+            }
+            if (systemStatus != null) {
+                updateState(TeslaPowerwallBindingConstants.CHANNEL_TESLAPOWERWALL_FULL_PACK_ENERGY,
+                        new QuantityType<>(systemStatus.full_pack_energy, Units.WATT_HOUR));
+                if (systemStatus.full_pack_energy < 13500) {
```

Comment:
13500 is a hardcoded value. all powerwalls have the same capacity?
Maybe better to extract this to a class level constant anyway.

Reply:
People with multiple powerwalls  will have > 13500 capacity -  right now I don't have sample data from someone with multiple powerwalls to understand how this information is presented via the API.

My intent is to support multiple powerwalls  via a future PR.

Reply:
Minor suggestion in that case: could you extract this to a binding constant?